test/js/node/watch/fs.watch.test.ts

```diff
@@ -97,6 +102,66 @@ describe("fs.watch", () => {
     });
   });
 
+  test("should watch new-created files", done => {
+    let count = 0;
+    const root = path.join(testDir, "add-directory");
+    try {
+      fs.mkdirSync(root);
+    } catch {}
+    let err: Error | undefined = undefined;
+    const watcher = fs.watch(root, { signal: AbortSignal.timeout(3000) });
+    let ready = false;
+    watcher.on("change", (event, filename) => {
+      count++;
+      try {
+        if (count == 1) {
+          ready = true;
+        } else if (count == 2) {
+          expect(event).toBe("rename");
+          expect(filename).toBe("new-file2.txt");
+        } else if (count == 3) {
+          expect(event).toBe("change");
+          expect(filename).toBe("new-file2.txt");
+        } else if (count == 4) {
+          expect(event).toBe("change");
+          expect(filename).toBe("new-file2.txt");
+        } else {
+          watcher.close();
+        }
+      } catch (e: any) {
+        err = e;
+        watcher.close();
+      }
+    });
+
+    watcher.on("error", e => (err = e));
+    watcher.on("close", () => {
+      clearInterval(interval);
+      if (err) {
+        done(err);
+        return;
+      }
+      try {
+        expect(count).toBeGreaterThan(3);
+      } catch (e: any) {
+        err = e;
+      }
+      done(err);
+    });
+
+    const interval = repeat(() => {
+      // Use this event to test if the watcher has been started.
+      if (!ready) {
+        fs.writeFileSync(path.join(root, "new-file.txt"), "hello");
+        return;
+      }
+      const fd = fs.openSync(path.join(root, "new-file2.txt"), "a");
+      fs.writeSync(fd, "hello");
+      fs.writeSync(fd, "world");
```

Comment:
I think I know why this test sometimes fails. Counting the number of events itself is unreliable, sometimes events get merged.

Reply:
Yeah `repeat` was used on another test to be more reliable